typing/typemod.ml

```diff
@@ -2194,8 +2194,8 @@ let wrap_constraint_with_shape env mark arg mty
   shape explicit =
   let coercion, shape =
     try
-      Includemod.modtypes_with_shape ~shape ~loc:arg.mod_loc env ~mark
-        arg.mod_type mty
+      Includemod.modtypes_with_shape ~shape ~loc:arg.mod_loc env
+        ~mark ~mark_both:true arg.mod_type mty
```

Comment:
I have the impression that it would be more regular to export a  `modtype_constraints` function in `Includemod` with an `unknown` direction (or export a direction argument in `modtypes_with_shape`)